e2e/nomostest/portforwarder/port_forwarder.go

```diff
@@ -233,24 +214,107 @@ func (pf *PortForwarder) start() error {
 	// Restart the port forward if the process exits prematurely (e.g. due to pod eviction)
 	go func() {
 		for {
-			if err := pf.watcher.WatchForNotFound(kinds.Pod(), pf.pod, pf.ns, testwatcher.WatchTimeout(0), testwatcher.WatchContext(pf.ctx)); err != nil {
-				if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
-					// stop retrying, the subprocess will be killed with the context.
-					return
-				}
-				// Watch automatically retries, so this shouldn't happen unless
-				// there's a terminal error.
-				pf.logger.Infof("Watch exited, restarting port-forward for %s/%s: %v", pf.ns, pf.deployment, err)
-			}
-			if err := pf.portForwardToDeployment(true); err != nil {
-				pf.logger.Info(err)
-				// this is a potentially fatal error since we failed to restart the port
-				// forward. we log the error and return in case the test is already done
-				// with the port forward. the test will otherwise error the next time it
-				// tries to use the local port.
+			if err := pf.waitAndRestart(); err != nil {
+				pf.logger.Infof("PortForwarder finished: %v", err)
 				return
 			}
 		}
 	}()
 	return nil
 }
+
+// waitAndRestart will block until seeing an indicator that the port-forward is
+// no longer healthy. If that occurs, it will restart the port-forward
+func (pf *PortForwarder) waitAndRestart() error {
+	err := pf.waitForFailure()
+	if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
+		return errors.Wrap(err, fmt.Sprintf("exiting - PortForwarder for pod %s/%s received done signal", pf.ns, pf.pod))
+	}
+	pf.logger.Infof("received signal for port-forward for %s/%s. err: %v", pf.ns, pf.pod, err)
```

Comment:
"signal" is a bit ambiguous to people who've never seen this code, especially if the error in nil. 
How about using different log msg for err vs nil?
Since we're not waiting for the watch and command to exit, and the watch always returns an error, we can be more specific:
`port-forward exited for pod %s/%s: %v`
else
`port-forward exited without error for pod %s/%s`